script/testing/self_driving/forecast.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+"""
+The forecast scripts generates metrics traces needed for workload forecasting.
+"""
+
+from oltpbench.test_case_oltp import TestCaseOLTPBench
+from oltpbench.run_oltpbench import TestOLTPBench
+from oltpbench.constants import OLTPBENCH_GIT_LOCAL_PATH
+from util.constants import (LOG, ErrorCode, DEFAULT_DB_HOST, DEFAULT_DB_PORT)
+from util.common import run_command
+from util.db_server import NoisePageServer
+from self_driving.constants import (
+    DEFAULT_DB_USER,
+    DEFAULT_TPCC_TIME_SEC,
+    DEFAULT_TPCC_WEIGHTS,
+    DEFAULT_QUERY_TRACE_FILE,
+    DEFAULT_OLTP_SERVER_ARGS,
+    DEFAULT_OLTP_TEST_CASE,
+    DEFAULT_WORKLOAD_PATTERN)
+
+import psycopg2 as psql
+from pathlib import Path
+from xml.etree import ElementTree
+from typing import Dict, List, Optional, Tuple
+
+
+def config_forecast_data(xml_config_file: str, rate_pattern: List[int]) -> None:
```

Comment:
Oh cool, didn't know that. What's the function you are referring to?

Reply:
`config_xml_file()`. It looks like its a class function but it could probably take in parameters instead of using `self.abc`. https://github.com/cmu-db/noisepage/blob/master/script/testing/oltpbench/test_case_oltp.py#L173